lib/rspec/support/object_inspector.rb

```diff
@@ -3,23 +3,51 @@ module Support
     # Provide additional output details beyond what `inspect` provides when
     # printing Time, DateTime, or BigDecimal
     module ObjectInspector
+      DEFAULT_OPTIONS = {
+        :nested => false,
+        :joiner => ',',
+        :braces => true,
+        :trailing_comma => false
+      }
       # @api private
-      def self.inspect(object)
-        if Time === object
+      # rubocop:disable CyclomaticComplexity
+      # rubocop:disable MethodLength
+      def self.inspect(object, options=DEFAULT_OPTIONS)
+        case object
+        when Time
           format_time(object)
-        elsif defined?(DateTime) && DateTime === object
-          format_date_time(object)
-        elsif defined?(BigDecimal) && BigDecimal === object
-          "#{object.to_s 'F'} (#{object.inspect})"
-        elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
-          object.description
+        when Hash
+          format_hash(object, options)
```

Comment:
@myronmarston This explanation might not be 100% correct;

`pp` does some stuff regarding formatting it's output, and the differ class seems to use some of the characters that it inserts in order to work out how to split up the array into "lines" before doing a diff on each line.

The output also goes a bit funny when total number of characters in an array that has been converted to a string, goes over a certain limit. 
If it's under a certain size, `pp` will output the whole array on one line. If it's over a certain size, it'll add `\n` to the end of each element and multiline it for you. I believe you can also control how many characters before it invokes this functionality with an environment variable (or something similar).

Anyway, in order to keep the differ working, either this functionality has to be recreated within the ObjectInspector, or the differ class gets a bit of a rework to work without expecting the string passed in to have already been formatted by `pp`.

I think its probably better for us to update the ObjectInspector AND the differ, to handle special objects inside arrays, in one pass, as I expect the differ class to get some changes, and the ObjectInspector to be extended with those changes in mind.

It would be much easier to implement if we just multilined every array when we output on the differ, but it makes sense for small arrays (that are easy to read), to be displayed on line.

I personally think it's much better for us leave the scope of this pull request to just deal with hashes, and create separate issue/pull-request that deals with arrays, in the near future.